src/VisualStudio/CSharp/Impl/ProjectSystemShim/CSharpProjectShim.ICSInputSet.cs

```diff
@@ -43,9 +44,14 @@ public void SetWin32Resource(string filename)
             // This file is used only during emit. Since we no longer use our in-proc workspace to emit, we can ignore this value.
         }
 
+        public CompilationOutputFiles CompilationOutputFiles
+            => (CompilationOutputFiles)VisualStudioProject.CompilationOutputs;
```

Comment:
would we ever expect VisualStudioProject to not be using a CompilationOutputFiles?

Reply:
So when redoing the project system code I was trying to keep VisualSudioProject fairly abstract and still making sure it doesn't assume:

1. That projects necessarily exist on disk.
2. They come from a particular source like MSBuild
3. They're C# or VB.

Right now ASP.NET is doing some work where they'll be creating projects that represent in-memory compilation results, and so theirs might be using the base type in this if we enable E&C for that. But for the legacy C#/VB shims that wrap all that, it's safe to assume. This cast makes total sense to me, other than wondering out loud if Tomas wants to move it to AbstractLegacyProject.